tests/e2e/network/service_annotations.go

```diff
@@ -315,3 +358,58 @@ func validateInternalLoadBalancer(c clientset.Interface, ns string, url string)
 	utils.Logf("validation finished")
 	return err
 }
+
+func (az *Test) validateLoadBalancerBackendPools(vmssname string, cs clientset.Interface, labels map[string]string, ns string, ports []v1.ServicePort, resourceGroupName string) {
+	
+	serviceName := "validate-lb-mode"
+	lbModeLoadBalancerName := serviceName + string(uuid.NewUUID())
+	// create deployment
+	utils.Logf("Creating deployment " + lbModeLoadBalancerName)
+	deployment := createNginxDeploymentManifest(lbModeLoadBalancerName, labels)
+	_, err := cs.AppsV1().Deployments(ns).Create(deployment)
+	Expect(err).NotTo(HaveOccurred())
+
+	// create annotation for LoadBalancer service
+	annotation := map[string]string{
+		azure.ServiceAnnotationLoadBalancerMode: vmssname,
+	}
+	service := createLoadBalancerServiceManifest(cs, lbModeLoadBalancerName, annotation, labels, ns, ports)
+	_, er := cs.CoreV1().Services(ns).Create(service)
+	Expect(er).NotTo(HaveOccurred())
+	utils.Logf("Successfully created LoadBalancer service " + lbModeLoadBalancerName + " in namespace " + ns)
+
+	//wait and get service's public IP Address
+	By("Waiting for service exposure")
+	pip, err := utils.WaitServiceExposure(cs, ns, serviceName)
+	Expect(err).NotTo(HaveOccurred())
+
+	//Invoking azure network client to get list of public IP Addresses
+	ctx := context.Background()
+	iplist, er := az.azutil.ListPublicIPs(ctx, resourceGroupName)
+	Expect(er).NotTo(HaveOccurred())
+	var pipID string
+	for _, ip := range iplist {
+		pipID = *ip.Properties.ipConfiguration.ID
+		if ip.PublicIPAddressPropertiesFormat != nil && *ip.PublicIPAddressPropertiesFormat.IPAddress == pip {
+			utils.Logf("Public Ip verified")
+		}
+	}
+	//Get Azure loadBalancer Name
+	match := lbNameRE.FindStringSubmatch(pipID)
+	if len(match) != 2 {
+		Skip("Report error")
+	}
+	loadBalancername := match[1]
+	//Get backendpools list
+	backendpools, err := az.client.ListComplete(ctx, resourceGroupName, loadBalancername)
+	Expect(err).NotTo(HaveOccurred())
+	//check whether vmss name equals the backendpool name
+
+	for ; backendpools.NotDone(); err = backendpools.Next() {
+		re := regexp.MustCompile(vmssname)
+		if nil != re.FindStringSubmatchIndex(*backendpools.Value().ID) {
+			Skip("Test passed for " + vmssname)
```

Comment:
shouldn't skip the tests here. We should filter out which backend pool is used by this publicIP (related by loadbalancer rule). And then, if the vmss in that backendpool is different, fail the test.